### PR TITLE
updated FunctionShield to 1.2.2

### LINKED ISSUE #252: New functionShield middleware results in warning from webpack
I'm not even using FunctionShield.

```
WARNING in ./node_modules/node-gyp-build/index.js
13:9-32 Critical dependency: the request of a dependency is an expression
 @ ./node_modules/node-gyp-build/index.js
 @ ./node_modules/@puresec/function-shield/dist/index.js
 @ ./node_modules/middy/src/middlewares/functionShield.js
 @ ./node_modules/middy/src/middlewares/index.js
 @ ./node_modules/middy/middlewares.js
 @ ./src/api-handler.ts

WARNING in ./node_modules/node-gyp-build/index.js
20:15-54 Critical dependency: the request of a dependency is an expression
 @ ./node_modules/node-gyp-build/index.js
 @ ./node_modules/@puresec/function-shield/dist/index.js
 @ ./node_modules/middy/src/middlewares/functionShield.js
 @ ./node_modules/middy/src/middlewares/index.js
 @ ./node_modules/middy/middlewares.js
 @ ./src/api-handler.ts
```

### PR DESCRIPTION
**Pull request by 0xh0b0**
Fixes #252

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -6396,9 +6396,8 @@
       "integrity": "sha1-KzJxhOiZIQEXeyhWP7XnECrNDKk="
     },
     "node-gyp-build": {
-      "version": "3.5.0",
-      "resolved": "https://registry.npmjs.org/node-gyp-build/-/node-gyp-build-3.5.0.tgz",
-      "integrity": "sha512-qjEE8eIWVyqZhkAFUzytGpOGvLHeX5kXBB6MYyTOCPZBrBlsLyXAAzTsp/hWMbVlg8kVpzDJCZZowIrnKpwmqQ=="
+      "version": "github:puresec/node-gyp-build#e75daf49c212b2190d2ccadb3c01c0ad0459d54b",
```
Comment (vladgolubev) [inline package-lock.json]: Diff of the actual fix: https://github.com/puresec/node-gyp-build/compare/master...puresec:dynamic_dep_fix#diff-168726dbe96b3ce427e7fedce31bb0bcR5

Comment (vladgolubev) [reply package-lock.json]: Is this something which can be fixed in the upstream repo so there is no need to maintain a fork?

Comment (0xh0b0) [reply package-lock.json]: I have created PR in the upstream repo: https://github.com/mafintosh/node-gyp-build/pull/10
We can wait for @mafintosh to merge it and publish a new package version.

Comment (0xh0b0) [reply package-lock.json]: We are now using an upstream version of node-gyp-build

Comment (vladgolubev) [reply package-lock.json]: @0xh0b0 thanks! That's awesome

**Comment by vladgolubev** [comment]
@0xh0b0 could you please bump the version to 0.19.5 here? https://github.com/middyjs/middy/blob/master/package.json#L3

Then I can merge the PR and get the new version published!

**Pull request closed without merging**